sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1177,6 +1180,54 @@ def test_large_regularization():
         assert_array_almost_equal(model.coef_, np.zeros_like(model.coef_))
 
 
+def test_tol_parameter():
+    # Test that the tol parameter behaves as expected
+    X = StandardScaler().fit_transform(iris.data)
+    y = iris.target
+    max_iter = 2000
+
+    # With tol == 0, the number of iteration should be equal to max_iter
+    model_0 = SGDClassifier(tol=0, max_iter=max_iter).fit(X, y)
+    assert_equal(max_iter, model_0.n_iter_)
+
+    # With tol > 0, the number of iteration should be less than max_iter
+    model_1 = SGDClassifier(tol=1e-2, max_iter=max_iter).fit(X, y)
+    assert_greater(max_iter, model_1.n_iter_)
+
+    # A larger tol should yield a smaller number of iteration
+    model_2 = SGDClassifier(tol=1e-1, max_iter=max_iter).fit(X, y)
+    assert_greater(model_1.n_iter_, model_2.n_iter_)
+
+    # A larger tol should yield a smaller number of iteration
```

Comment:
This tests lasts 50 ms, removing `model_3` does not speed it up since there is few iterations compared to `model_0` or `model_1`.

I changed `model_0` to go down to 20 ms.

The other tests lasts 9 ms.